{
  "modified": "2020-01-21T21:06:20Z",
  "published": "2012-08-27T23:00:00Z",
  "schema_version": "1.5.0",
  "id": "PSF-2012-2",
  "aliases": [
    "CVE-2011-4944"
  ],
  "summary": "pypirc created insecurely",
  "details": "Python 2.6 through 3.2 creates ~/.pypirc with world-readable permissions before changing them after data has been written, which introduces a race condition that allows local users to obtain a username and password by reading this file.",
  "affected": [
    {
      "ranges": [
        {
          "type": "GIT",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "d61926e6bef6c4d8105a2848362377dce91d7fc8"
            },
            {
              "fixed": "e5567ccc863cadb68f5e57a2760e021e0d3807cf"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "database_specific": {
    "cwe_ids": []
  },
  "references": [
    {
      "type": "REPORT",
      "url": "https://bugs.python.org/issue13512"
    }
  ]
}